### ISSUE TITLE
Doc lacking

### ISSUE THREAD

**Issue by kasuteru** [Azure/azure-sdk-for-python#16163]
This could really use more information, e.g.:

1. Do I have to specify data? Or can I just leave it at none & have Azure figure it out?
2. Data is supposed to be binary I guess? How does it treat strings?
3. An example... I am getting confusing errors
4. A note that files are not automatically created and have to be created first. This is in contrast to Python's usual way of writing to files, file.open(xx, "w").

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 50acce08-5aa7-19f6-a652-d8606ad7595d
* Version Independent ID: 5399f9d1-65df-b907-aee4-69584cc3e7f1
* Content: [azure.storage.filedatalake.DataLakeFileClient class](https://docs.microsoft.com/en-us/python/api/azure-storage-file-datalake/azure.storage.filedatalake.datalakefileclient?view=azure-python)
* Content Source: [docs-ref-autogen/azure-storage-file-datalake/azure.storage.filedatalake.DataLakeFileClient.yml](https://github.com/MicrosoftDocs/azure-docs-sdk-python/blob/master/docs-ref-autogen/azure-storage-file-datalake/azure.storage.filedatalake.DataLakeFileClient.yml)
* GitHub Login: @lmazuel
* Microsoft Alias: **lmazuel**

**Comment by tasherif-msft** [comment]
Hi @kasuteru,
Could you please be more specific on your requests? there is very little context but I will address your points as best I can:
1. Specify data where? if you're referring to instantiating a file client, you do not provide data. A file client simply gives you a way to interact with the file object that has been created or about to be created. Using helper methods you can upload/append data.
2. Appended/uploaded Data can be strings.
3. Confusing errors on which operation?
4. The DatalakeFileClient states the following in its docstrings "A client to interact with the DataLake file, even if the file may not yet exist."

**Comment by kasuteru** [comment]
Hi,

sorry about that, I admit I wasn't very specific on that. To go more into detail:

1. I think I made a typo here. I actually wanted to ask about the "length" parameter in `append_data()`. The default is none, why does this parameter exist? Isn't the length of the data already implied? And can I just leave it at None, what happens if I don't? Is part of data thrown away if length is lower? What happens if length is more than the data provided?
2. If I provide a string, what encoding will woould be used? As to my understanding, in the end, it will somehow have to be binary. Maybe a bit more explanation what happens when type(data) == str would help
3. I actually don't remember this, sorry about that.
4. It does state that, but I initially thought that getting the Client is enough to create the file, before I saw that there is an explicit create_file() method. Maybe a sentence about how constructing a DataLakeFileClient instance does not automatically create the underlying file would help future readers (I now it know, but others might make the same mistake as I did)
